model_training/run.py

```diff
@@ -23,13 +26,46 @@ def get_config(config_path: Path = Path("config/config.yaml")) -> Any:
     return read_yaml_file(config_path)
 
 
+def get_train_data_version(root_git_dir: str = "..") -> str | None:
+    """Get git tag corresponding to training_data.dvc file.
+
+    Args:
+        root_git_dir: Path to root of the git repo
+
+    Returns:
+        Git tag corresponding to training data
+
+    Raises:
+        Exception is thrown if tag is not found for training_data.dvc file
+    """
+    repo = Repo(root_git_dir)
+    current_commit, _ = repo.blame("HEAD", file="model_training/training_data.dvc")[0]
+    logger.debug(
+        f"Commit corresponding to 'model_training/training_data.dvc' file: {current_commit}"
+    )
+    git_cmd = repo.git
+    try:
+        current_tag = git_cmd.tag(current_commit, contains=True)
+        logger.debug(
+            f"Tag corresponding to 'model_training/training_data.dvc' file: {current_tag}"
+        )
+        return current_tag
+    except Exception as e:
+        logger.warning(
+            "No tag found for current 'model_training/training_data.dvc' file\n"
+            f"Command failed with following error : {e}"
+        )
+        return None
```

Comment:
I would say that if there's no tag, then commit hash is a valid data version.

Reply:
Added commit as fallback.